packages/fluentui/docs/src/prototypes/VirtualizedStickyTree/VirtualizedStickyTree.tsx

```diff
@@ -0,0 +1,471 @@
+import { InnerElementContext, InnerElementContextType } from './context';
+
+class ItemWrapper extends React.PureComponent<{
+  index: number;
+  isScrolling?: boolean;
+  style: Object;
+  data: any;
+}> {
+  render() {
+    const { index, style, data } = this.props;
+    const item = data.renderedItems[index];
+    if (item) {
+      return data.isNonSticky(item)
+        ? React.cloneElement(item, {
+            style,
+            onTitleClick: data.onClick,
+            onFocusParent: data.handleFocusParent,
+            onFocusFirstChild: data.handleFocusFirstChild,
+            onSiblingsExpand: (event: React.SyntheticEvent<HTMLElement>, treeItemProps: TreeItemProps) => {
+              if (treeItemProps.level !== 1)
+                // don't care about onsibling Expand for sticky headers
+                data.handleOnSiblingsExpand(event, treeItemProps);
+            },
+            onFocus: (e: React.FocusEvent<HTMLElement>) => {
+              if (item.props.level === 1) return;
+              data.handleOnFocus(e);
+            },
+          })
+        : null;
+    }
+
+    return null;
+  }
+}
+
+const InnerElementType = ({ children, style }) => {
+  const context = React.useContext(InnerElementContext);
+  const { height, stickyItemSize, stickyTopIds, stickyBottomIds, onClickSticky } = context;
+
+  if (!children?.length) return null;
+
+  const data = children[0].props?.data;
+  const { renderedItems } = data;
+
+  const stickyTopItemsIndex = stickyTopIds.map((id: string) =>
+    renderedItems.findIndex((item: React.ReactElement<TreeItemProps>) => item.props.id === id),
+  );
+  const stickyBottomItemsIndex = stickyBottomIds.map((id: string) =>
+    renderedItems.findIndex((item: React.ReactElement<TreeItemProps>) => item.props.id === id),
+  );
+
+  return (
+    <div
+      style={{
+        ...style,
+        height: Math.max(height, style.height),
+      }}
+    >
+      {stickyTopItemsIndex.map((index: number) => {
+        const item = renderedItems[index];
+        return React.cloneElement(item, {
+          style: {
+            position: 'sticky',
+            height: stickyItemSize,
+            zIndex: teamsTheme.siteVariables.zIndexes.overlay,
+            top: index * stickyItemSize,
+          },
+          onTitleClick: onClickSticky,
+          onFocusParent: data.handleFocusParent,
+          onFocusFirstChild: data.handleFocusFirstChild,
+        });
+      })}
+      {children}
+      {stickyBottomItemsIndex.map((itemIndex: number, index: number, arr: number[]) => {
+        const item = renderedItems[itemIndex];
+        return React.cloneElement(item, {
+          style: {
+            position: 'sticky',
+            height: stickyItemSize,
+            zIndex: teamsTheme.siteVariables.zIndexes.overlayPriority,
+            // stick items to the container bottom, with the smaller-indexed items stacked on bigger-indexed items
+            top: height - (arr.length - index) * stickyItemSize,
```

Comment:
nit: you could generate this obj in a function and pass in the value for top to avoid duplicating the other styles with the top stickies